ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/dot1x.j2

```diff
@@ -79,5 +99,29 @@ dot1x
    radius av-pair framed-mtu {{ dot1x.radius_av_pair.framed_mtu }}
 {%             endif %}
 {%         endif %}
+{%         if dot1x.supplicant.disconnect_cached_results_timeout is arista.avd.defined %}
+   supplicant disconnect cached-results timeout {{ dot1x.supplicant.disconnect_cached_results_timeout }} seconds
+{%         endif %}
+{%         if dot1x.captive_portal.url is arista.avd.defined or dot1x.captive_portal.ssl_profile is arista.avd.defined %}
+{%             set captive_portal_cli = "captive-portal" %}
+{%             if dot1x.captive_portal.url is arista.avd.defined %}
+{%                 set captive_portal_cli = captive_portal_cli ~ " url " ~ dot1x.captive_portal.url %}
+{%             endif %}
+{%             if dot1x.captive_portal.ssl_profile is arista.avd.defined %}
+{%                 set captive_portal_cli = captive_portal_cli ~ " ssl profile " ~ dot1x.captive_portal.ssl_profile %}
+{%             endif %}
+   {{ captive_portal_cli }}
+{%         elif dot1x.captive_portal.enabled is arista.avd.defined(true) %}
+   captive-portal
+{%         endif %}
```

Comment:
```suggestion
{%         if dot1x.captive_portal.enabled is arista.avd.defined(true) %}
{%             set captive_portal_cli = "captive-portal" %}
{%             if dot1x.captive_portal.url is arista.avd.defined %}
{%                 set captive_portal_cli = captive_portal_cli ~ " url " ~ dot1x.captive_portal.url %}
{%             endif %}
{%             if dot1x.captive_portal.ssl_profile is arista.avd.defined %}
{%                 set captive_portal_cli = captive_portal_cli ~ " ssl profile " ~ dot1x.captive_portal.ssl_profile %}
{%             endif %}
   {{ captive_portal_cli }}
{%         endif %}
```

Reply:
Are you sure you can give ssl profile without a URL?

Reply:
yes, it works
```
s1-leaf2(config-dot1x)#captive-portal ssl profile sdf 
s1-leaf2(config-dot1x)#show active
dot1x
   captive-portal ssl profile sdf
```

Reply:
Re- the above suggestion, I implemented the same earlier but `enabled` `url` and `ssl` are mutually exclusive
```
s1-leaf2(config-dot1x)#captive-portal ssl profile sdf 
s1-leaf2(config-dot1x)#show active
dot1x
   captive-portal ssl profile sdf
s1-leaf2(config-dot1x)#captive-portal ?
  access-list  Configure access control list
  ssl          Configure SSL related options
  url          Configure captive portal URL
  <cr>         

s1-leaf2(config-dot1x)#captive-portal 
s1-leaf2(config-dot1x)#show active
dot1x
   captive-portal
s1-leaf2(config-dot1x)#captive-portal url http://sdcg:24 ssl profile pf1
s1-leaf2(config-dot1x)#show active
dot1x
   captive-portal url http://sdcg:24 ssl profile pf1
```

Reply:
My point is to make `enabled` the "guard" for configuring anything. That is how we do it elsewhere in eos_cli_config_gen.
So if you _only_ get `enabled: true` you configure just the `captive-portal` command, and for every option you set, you add to this keyword the options accordingly.

Reply:
Okay, I have updated the template as per suggestion